// react-js-snackbar/lib/index.js
"use strict";

exports.__esModule = true;

var _templateObject = _taggedTemplateLiteralLoose(["\n  visibility: visible;\n  max-width: 50px;\n  height: 50px;\n  margin: auto;\n  background-color: #333;\n  color: #fff;\n  text-align: center;\n  position: fixed;\n\n  z-index: 1;\n  left: 0;\n  right: 0;\n  bottom: 30px;\n  font-size: 17px;\n  white-space: nowrap;\n  border-radius: 24px;\n  text-overflow: ellipsis;\n\n  ", ";\n  ", ";\n\n  ", ";\n\n  ", ";\n\n  @keyframes fadein {\n    from {\n      bottom: 2px;\n      opacity: 0;\n    }\n    to {\n      bottom: 30px;\n      opacity: 1;\n    }\n  }\n\n  @keyframes expand {\n    from {\n      opacity: 1;\n      min-width: 50px;\n    }\n    to {\n      opacity: 1;\n      min-width: 350px;\n    }\n  }\n\n  @keyframes stay {\n    from {\n      opacity: 1;\n      min-width: 350px;\n    }\n    to {\n      opacity: 1;\n      min-width: 350px;\n    }\n  }\n\n  @keyframes shrink {\n    from {\n      opacity: 1;\n      min-width: 350px;\n    }\n    to {\n      opacity: 1;\n      min-width: 50px;\n    }\n  }\n\n  @keyframes fadeout {\n    from {\n      min-width: 50px;\n      bottom: 30px;\n      opacity: 1;\n    }\n    to {\n      min-width: 50px;\n      bottom: 60px;\n      opacity: 0;\n    }\n  }\n"], ["\n  visibility: visible;\n  max-width: 50px;\n  height: 50px;\n  margin: auto;\n  background-color: #333;\n  color: #fff;\n  text-align: center;\n  position: fixed;\n\n  z-index: 1;\n  left: 0;\n  right: 0;\n  bottom: 30px;\n  font-size: 17px;\n  white-space: nowrap;\n  border-radius: 24px;\n  text-overflow: ellipsis;\n\n  ", ";\n  ", ";\n\n  ", ";\n\n  ", ";\n\n  @keyframes fadein {\n    from {\n      bottom: 2px;\n      opacity: 0;\n    }\n    to {\n      bottom: 30px;\n      opacity: 1;\n    }\n  }\n\n  @keyframes expand {\n    from {\n      opacity: 1;\n      min-width: 50px;\n    }\n    to {\n      opacity: 1;\n      min-width: 350px;\n    }\n  }\n\n  @keyframes stay {\n    from {\n      opacity: 1;\n      min-width: 350px;\n    }\n    to {\n      opacity: 1;\n      min-width: 350px;\n    }\n  }\n\n  @keyframes shrink {\n    from {\n      opacity: 1;\n      min-width: 350px;\n    }\n    to {\n      opacity: 1;\n      min-width: 50px;\n    }\n  }\n\n  @keyframes fadeout {\n    from {\n      min-width: 50px;\n      bottom: 30px;\n      opacity: 1;\n    }\n    to {\n      min-width: 50px;\n      bottom: 60px;\n      opacity: 0;\n    }\n  }\n"]),
    _templateObject2 = _taggedTemplateLiteralLoose(["\n      visibility: visible;\n      animation: fadein 0.5s, expand 0.5s 0.5s, stay 99999999995s 1s;\n      /* shrink 1s 5s,\n        fadeout 0.5s 6s; */\n    "], ["\n      visibility: visible;\n      animation: fadein 0.5s, expand 0.5s 0.5s, stay 99999999995s 1s;\n      /* shrink 1s 5s,\n        fadeout 0.5s 6s; */\n    "]),
    _templateObject3 = _taggedTemplateLiteralLoose(["\n      min-width: 350px;\n    "], ["\n      min-width: 350px;\n    "]),
    _templateObject4 = _taggedTemplateLiteralLoose(["\n      visibility: visible;\n      animation: shrink 0.5s 0.5s, fadeout 1s 1s;\n    "], ["\n      visibility: visible;\n      animation: shrink 0.5s 0.5s, fadeout 1s 1s;\n    "]),
    _templateObject5 = _taggedTemplateLiteralLoose([""], [""]),
    _templateObject6 = _taggedTemplateLiteralLoose(["\n  border-radius: 24px;\n  width: 50px;\n  height: 50px;\n\n  float: left;\n\n  padding-top: 14px;\n\n  box-sizing: border-box;\n  background-color: #111;\n  color: #fff;\n"], ["\n  border-radius: 24px;\n  width: 50px;\n  height: 50px;\n\n  float: left;\n\n  padding-top: 14px;\n\n  box-sizing: border-box;\n  background-color: #111;\n  color: #fff;\n"]),
    _templateObject7 = _taggedTemplateLiteralLoose(["\n  color: white;\n  transition: all 150ms linear;\n"], ["\n  color: white;\n  transition: all 150ms linear;\n"]),
    _templateObject8 = _taggedTemplateLiteralLoose(["\n  color: #fff;\n  align-self: center;\n  padding-left: 12px;\n  overflow: hidden;\n  white-space: nowrap;\n  text-align: left;\n  text-overflow: ellipsis;\n\n  text-align: center;\n  vertical-align: middle;\n"], ["\n  color: #fff;\n  align-self: center;\n  padding-left: 12px;\n  overflow: hidden;\n  white-space: nowrap;\n  text-align: left;\n  text-overflow: ellipsis;\n\n  text-align: center;\n  vertical-align: middle;\n"]),
    _templateObject9 = _taggedTemplateLiteralLoose(["\n  transition: all 150ms linear;\n  display: flex;\n  height: 100%;\n  position: relative;\n\n  overflow: hidden;\n  text-overflow: ellipsis;\n  white-space: nowrap;\n"], ["\n  transition: all 150ms linear;\n  display: flex;\n  height: 100%;\n  position: relative;\n\n  overflow: hidden;\n  text-overflow: ellipsis;\n  white-space: nowrap;\n"]);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require("styled-components");

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _reactTransitionGroup = require("react-transition-group");

var _propTypes = require("prop-types");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteralLoose(strings, raw) { strings.raw = raw; return strings; }

var MainContainer = _styledComponents2.default.div(_templateObject, function (props) {
  return props.Show && (0, _styledComponents.css)(_templateObject2);
}, function (props) {
  return !props.Show && (0, _styledComponents.css)(_templateObject3);
}, function (props) {
  return props.state === "exiting" && (0, _styledComponents.css)(_templateObject4);
}, function (props) {
  return props.state === "exited" && (0, _styledComponents.css)(_templateObject5);
});

var IconContainer = _styledComponents2.default.div(_templateObject6);

var DefaultIcon = _styledComponents2.default.i(_templateObject7);

var DefaultText = _styledComponents2.default.div(_templateObject8);

var TextContainer = _styledComponents2.default.div(_templateObject9);
var ReactSnackBar = function ReactSnackBar(props) {
  return _react2.default.createElement(
    _reactTransitionGroup.CSSTransition,
    {
      "in": props.Show,
      classNames: "_bottom-toaster-transition",
      timeout: props.Show ? 0 : 1500,
      unmountOnExit: true
    },
    function (state) {
      return _react2.default.createElement(
        MainContainer,
        { state: state, Show: props.Show },
        _react2.default.createElement(
          IconContainer,
          null,
          _react2.default.createElement(
            DefaultIcon,
            null,
            props.Icon
          )
        ),
        _react2.default.createElement(
          TextContainer,
          null,
          _react2.default.createElement(
            DefaultText,
            null,
            props.children
          )
        )
      );
    }
  );
};

ReactSnackBar.propTypes = process.env.NODE_ENV !== "production" ? {
  children: _propTypes.PropTypes.any,
  Icon: _propTypes.PropTypes.any,
  Show: _propTypes.PropTypes.bool
} : {};
exports.default = ReactSnackBar;
module.exports = exports["default"];
